torch/quantization/convert_modules.py

```diff
@@ -0,0 +1,222 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+import torch.nn as nn
+import torch.nn.quantized as nnq
+import torch
+from .observer import *
+import heapq
+
+def _forward_hook(self, input, output):
+    for module in self.children():
+        module.forward(output)
+
+def add_observer(module, qconfig_dict, qconfig_parent=None, prefix=''):
+    r"""Transform a module to a quantized module according to qconfig_dict
+
+    This function insert observer module to all leaf child module of a given
+    module based on qconfig_dict.
+
+    Args:
+        module: instance of the module we want to transform
+        qconfig_dict: dictionary that maps from name of submodule to quantization
+                     configuration
+        qconfig_parent: quantization config of parent module, we will fallback to
+                       this config when there is no specified config for current
+                       module
+        prefix: corresponding prefix of the current module, used as key in
+                qconfig_dict
+    """
+    assert not hasattr(module, 'qconfig'), 'Original module should not have \
+        qconfig object already'
+    module.qconfig = None
+    if prefix in qconfig_dict:
+        module.qconfig = qconfig_dict[prefix]
+    else:
+        module.qconfig = qconfig_parent
+
+    # Insert observers only for leaf nodes, note that this observer is for
+    # the output of the module, for input QuantStub will observe them
+    if module.qconfig is not None and len(module._modules) == 0:
```

Comment:
no need to access `_modules`, do `len(module.children())`

Reply:
not addressed

Reply:
I checked this one, `module.children()` is an iterator, we'll have to do `len(list(module.children())` should we do that instead of accessing `_modules`?